keps/sig-storage/3476-volume-group/README.md

```diff
@@ -0,0 +1,1680 @@
+// VolumeGroupSpec describes the common attributes of group storage devices
+// and allows a Source for provider-specific attributes
+Type VolumeGroupSpec struct {
+        // +optional
+        VolumeGroupClassName *string
+
+        // If ImmutableSource is nil, an empty volume group will be created.
+        // Otherwise, a volume group will be created with PVCs (if PVCList or Select is set)
+        // If ImmutableSource is not nil, it indicates the VolumeGroup is immutable
+        // +optional
+        ImmutableSource *VolumeGroupSource
+}
+
+// VolumeGroupSource contains 3 options. If VolumeGroupSource is not nil,
+// one of the 3 options must be defined.
+Type VolumeGroupSource struct {
+        // A list of existing persistent volume claims
+        // +optional
+        PVCList []PersistentVolumeClaim
+
+        // A label query over existing persistent volume claims to be added to the volume group.
+        // +optional
+        Selector *metav1.LabelSelector
+ }
+
+type VolumeGroupStatus struct {
+        // VolumeGroupId is a unique id returned by the CSI driver
+        // to identify the VolumeGroup on the storage system.
+        // If a storage system does not provide such an id, the
+        // CSI driver can choose to return the VolumeGroup name.
+        VolumeGroupId *string
+
+        GroupCreationTime *metav1.Time
+
+        // A list of persistent volume claims
+        // +optional
+        PVCList []PersistentVolumeClaim
+
+        Ready *bool
+
+        // Mutable indicates if a VolumeGroup can be modified
+        // after it is created. If false, it indicates it cannot be
+        // modified once created. If ImmutableSource is not nil
+        // in VolumeGroupSpec, Mutable must be false; otherwise
+        // it means the driver does not support ImmutableSource.
+        // VOLUMEGROUP_IMMUTABLE and VOLUMEGROUP_MUTABLE capability
+        // will be added to the CSI spec.
+        Mutable *bool
+
+        // If true, it indicates the CSI driver supports adding
+        // an existing volume to the VolumeGroup and removing a
+        // volume from the VolumeGroup without deleting it.
+        // Only mutable VolumeGroup can support AddRemoveExistingPVC.
+        // A corresponding VOLUMEGROUP_ADD_REMOVE_EXISTING_VOLUME
+        // capability will be added to the CSI spec.
+        AddRemoveExistingPVC *bool
+
+        // Last error encountered during group creation
+        Error *VolumeGroupError
+}
+```
+
+VOLUMEGROUP_IMMUTABLE and VOLUMEGROUP_MUTABLE capability will be added to the CSI spec.
+If VOLUMEGROUP_IMMUTABLE is supported, a VolumeGroup with an ImmutableSource can be created. Mutable will be false, PVCList will be set, and Ready will be true in the Status.
+Otherwise, a VolumeGroup with an ImmutableSource will not be created successfully.
+
+## Proposal for Volume Placement
+
+### API Changes
+
+In order to support Volume Placement, An `AllowedTopologies` field will be added to the VolumeGroupClass API:
```

Comment:
Is this sufficient? I think the most common use case would be to spread volumes across different topologies.

Since we stated above that this is out of scope of the kep, maybe we should remove this and park it somewhere else.